src/components/RightSidebar/RightSidebar.astro

```diff
@@ -37,10 +32,18 @@ const headers = content.astro?.headers;
 		position: sticky;
 		top: 0;
 	}
+
 	.sidebar-nav-inner {
 		height: 100%;
 		padding: 0;
 		padding-top: var(--doc-padding);
 		overflow: auto;
 	}
-</style>
+
+	/* style applied only on Firefox, this avoids scrollbars overlapping. */
+	@supports (-moz-appearance: none) {
+		.sidebar-nav {
+			padding-inline-end: 1.5rem;
+		}
+	}
+</style>
```

Comment:
Looks good!

I tested in Safari and it seems to follow Chromium in automatically adding padding around scrollbars.

I also tried adding this universally but it’s visually pretty unsatisfactory as you get loads of extra space in odd places, so I think your solution is probably best! 🙌